pkg/populate/populate.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 OnMetal authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package populate
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/go-logr/logr"
+	onmetalimage "github.com/onmetal/onmetal-image"
+	"github.com/onmetal/onmetal-image/oci/remote"
+)
+
+func ResolveImage(ctx context.Context, log logr.Logger, imageName string) (*onmetalimage.Image, error) {
+	log.V(1).Info("Resolving image", "name", imageName)
+	reg, err := remote.DockerRegistry(nil)
```

Comment:
We shouldn't instantiate the `remote.DockerRegistry` for each image resolution call - some entity should store the result of this. Maybe another thing to move to a potential image fetching entity?